internal/controller/git/base_repo.go

```diff
@@ -232,13 +232,20 @@ func (b *baseRepo) buildCommand(command string, arg ...string) *exec.Cmd {
 func (b *baseRepo) buildGitCommand(arg ...string) *exec.Cmd {
 	cmd := b.buildCommand("git", arg...)
 	cmd.Env = append(cmd.Env, fmt.Sprintf("GIT_SSH_COMMAND=ssh -F %s/.ssh/config", b.homeDir))
+
 	if b.creds != nil && b.creds.Password != "" {
-		cmd.Env = append(
-			cmd.Env,
-			"GIT_ASKPASS=/usr/local/bin/credential-helper",
-			fmt.Sprintf("GIT_PASSWORD=%s", b.creds.Password),
-		)
+		cmd.Env = append(cmd.Env, "GIT_ASKPASS=/usr/local/bin/credential-helper")
+		cmd.Env = append(cmd.Env, fmt.Sprintf("GIT_PASSWORD=%s", b.creds.Password))
+	}
+
+	if httpProxy := os.Getenv("HTTP_PROXY"); httpProxy != "" {
+		cmd.Env = append(cmd.Env, fmt.Sprintf("http_proxy=%s", httpProxy))
 	}
+
+	if httpsProxy := os.Getenv("HTTPS_PROXY"); httpsProxy != "" {
+		cmd.Env = append(cmd.Env, fmt.Sprintf("https_proxy=%s", httpsProxy))
+	}
```

Comment:
`http_proxy` and `https_proxy` are well-known env vars respected by many tools. I'd prefer that we use those (lower case) instead of `HTTP_PROXY` and `HTTPS_PROXY`. i.e. If they're defined, just copy them as-is into the command's env.

It would probably be a good idea to make these configurable via the chart as well.